src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ListingService.java

```diff
@@ -312,6 +315,23 @@ private ListingData getDateRangeReport(ListingDetails listingDetails, String aut
         }
     }
 
+    private void checkForExistingData(ListingData listingData) {
```

Comment:
There are a lot of if conditions in the approach taken. Is this unnecessarily complicated? Couldn't we just clear those fields every time?

Reply:
I suppose we could clear them each time